### PR TITLE
Add support for bundling trace server

### LINKED ISSUE #17: Bundle Trace Compass server with Theia Blueprint
For the TraceViewer extension to work we need the trace compass server. At the moment the user has to download and run the server themselves. Instead we should just bundle it together with the Electron application [similar to the Trace Viewer example application](https://github.com/eclipse-cdt-cloud/theia-trace-extension/blob/0d943996bac57345a367d1b70e6ccfbc144d55d3/examples/electron/electron-builder.yml).

### PR DESCRIPTION
**Pull request by bhufmann**
#### What it does

1. Package the trace server in the CDTCloudBlueprint.AppImage for Linux (fixes #17)
2. Support running CDT cloud blueprint from development environment (both browser and electron app). Set trace server path for that.

#### How to test

1. Run `yarn tracecompass-server:download` first and then package the application image on Linux. Start the application and use command `Start trace server` to start trace server. The server should start successfully.
2. `yarn browser` and connect with browser to localhost:3000 or `yarn electron`. In the running application use command `Start trace server` to start trace server. The server should start successfully.

Note, that this PR also includes a commit to Workaround rate limitation issues with open-vsx. It downloads the plug-ins sequentially instead of parallel. This should avoid the failure "Download:plugins fails with 429". See discussion:
https://community.theia-ide.org/t/download-plugins-fails-with-429/2659

### CODE REVIEW

**Comment by bhufmann** [comment]
@sdirix the build fails. It seems unrelated but I'm not sure. Any suggestion?

File: package.json [hunk 1]
```diff
@@ -52,8 +52,8 @@
     "bundle": "lerna run bundle",
     "watch": "lerna run --parallel watch",
     "test": "lerna run test",
-    "electron": "yarn --cwd applications/electron",
-    "browser": "yarn --cwd applications/browser",
```
Comment (sdirix) [inline package.json:56]: Those should not be adapted. The idea is to call `yarn electron start` in the root directory

**Review by sdirix — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution :heart: ! I did not check too deeply into the failing builds yet but could be related to the contained script change.

If that's not it I can take a look early next week.

File: package.json (hunk 1 above)
Comment (bhufmann) [reply package.json:56]: Thanks for the comment. I'll upload a new version of the PR.

**Review by sdirix — COMMENTED** [review]
Works for me with the packaged Electron AppImage! Just a minor comment regarding the automatic build/publish flow.

Do you know how this works in the packaged Electron build? Is the trace viewer extension smart enough to check the bundled resources?

File: applications/electron/electron-builder.yml [hunk 2]
```diff
@@ -51,6 +51,9 @@ linux:
   publish:
     provider: generic
     url: "https://download.eclipse.org/theia/cdt-cloud/latest/linux"
+  extraResources:
+    - from: ../../tracecompass-server/trace-compass-server
```
[the remark this answers is not in the document]
Comment (bhufmann) [reply applications/electron/electron-builder.yml:55]: I'll update that.

**Comment by bhufmann** [comment]
> Works for me with the packaged Electron AppImage! Just a minor comment regarding the automatic build/publish flow.
> 
> Do you know how this works in the packaged Electron build? Is the trace viewer extension smart enough to check the bundled resources?

Yes, the extension code is looking into the resource directory of the AppImage for the executable of the server (untill a user changes the preferences).

**Comment by sdirix** [comment]
@bhufmann I added a commit how I would like to see the download of the trace viewer to be handled. if you agree with this approach I would squash my change into your commit and merge.

**Comment by bhufmann** [comment]
> @bhufmann I added a commit how I would like to see the download of the trace viewer to be handled. if you agree with this approach I would squash my change into your commit and merge.

Thanks for providing the clarification. I'll update the PR and integrate your change.

**Pull request merged**